### PR TITLE
Bug fixes for 0.16 release

### PR DESCRIPTION
**Pull request by Kucharssim**
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1424
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1425
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1426
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1427
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1428
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1429
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1430

and added tests where appropriate

### CODE REVIEW

**Comment by Kucharssim** [comment]
Eh, the test is failing on Windows where I compare whether the analysis returns a correct error message (that was compoased using `gettextf` with a unicode character). Not sure this should be done properly, do you have suggestions @vandenman?

File: tests/testthat/test-onesample.R [hunk 1]
```diff
@@ -36,4 +38,22 @@ test_that("Von Mises Assumption Check table results match", {
...
+  options$period <- 360
+  options$periodGroup <- "custom"
+  options$variables <- "contWide"
+  options$vonMisesCheck <- TRUE
+  
+  set.seed(1)
+  results <- runAnalysis("CircularStatisticsOneSampleTests", "test.csv", options)
+  errorMessage <- results[["results"]][["vonMisesCheckTable"]][["error"]][["errorMessage"]]
+  testthat::expect_identical(errorMessage, 
+                             gettextf("Estimated %s is infinite, could not compute results. Your data is too concentrated to calculate the assumption check.", "\u03BA"))
```
Comment (vandenman) [inline tests/testthat/test-onesample.R:58]: I don't really know how to fix this properly, it should just work the way you did it. We could manipulate the unicode with stringi/ stringr, but I think it's more appropriate to do
```r
if (.Platform$OS.type == "windows") {
  testthat::expect_true(grepl("Estimated \\S is infinite, could not compute results. Your data is too concentrated to calculate the assumption check.", errorMessage))
} else {
  testthat::expect_identical(errorMessage, 
                             gettextf("Estimated %s is infinite, could not compute results. Your data is too concentrated to calculate the assumption check.", "\u03BA"))
}
```
Here `\\S` is a catch-all for a single non-whitespace character.

**Review by vandenman — CHANGES_REQUESTED** [review]
Good call on the common QML element. I proposed a fix for the unit test below and I'll also check the issues in JASP now.

**Comment by vandenman** [comment]
All issues are fixed btw.

- [x] jasp-stats/INTERNAL-jasp#1424
- [x] jasp-stats/INTERNAL-jasp#1425
- [x] jasp-stats/INTERNAL-jasp#1426
- [x] jasp-stats/INTERNAL-jasp#1427
- [x] jasp-stats/INTERNAL-jasp#1428
- [x] jasp-stats/INTERNAL-jasp#1429
- [x] jasp-stats/INTERNAL-jasp#1430

File: tests/testthat/test-onesample.R (hunk 1 above)
Comment (Kucharssim) [reply tests/testthat/test-onesample.R]: Thanks! Let's hope that the test will now pass

**Review by vandenman — APPROVED** [review]
If the tests pass this can be merged. @Kucharssim will you also open a PR to jasp-desktop to reinclude the module or shall I do that?

**Comment by Kucharssim** [comment]
> If the tests pass this can be merged. @Kucharssim will you also open a PR to jasp-desktop to reinclude the module or shall I do that?

I can do that; I will also add a circular file to Data Library while at it

**Comment by vandenman** [comment]
hmm, the test failed again... perhaps just skip the test on windows for now? I don't have access to a windows machine right now and it feels like a waste of time to really debug this since it's likely just the same old issue where R doesn't handle Unicode properly on windows.

**Pull request merged**